solutions/arrays/reduce-array.js

```diff
@@ -0,0 +1,32 @@
+let fs = require("fs");
+let data = fs.readFileSync(0, 'utf-8');
+let idx = 0;
+data = data.split('\n');
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+// Use readLine() for taking input, it will read one line of from the input  and is stored in string format
+let arraySize=parseInt(readLine());
+let arr=[];
+for(let index=0;index<arraySize;index++){
+	arr.push(parseInt(readLine()));

```

Comment:
@DevratAgarwal We don't need to store all the elements. We can do it with O(1) space.